eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -302,6 +302,7 @@ private[channel] object ChannelCodecs3 {
         ("buried" | bool8) ::
         ("channelAnnouncement" | optional(bool8, lengthDelimited(channelAnnouncementCodec))) ::
         ("channelUpdate" | lengthDelimited(channelUpdateCodec)) ::
+        ("channelUpdateBeforeRestore_opt" | provide[Option[ChannelUpdate]](None)) ::
```

Comment:
Unfortunately it's not that simple, because what you're doing here means you're never actually saving it in the DB, so it will always be `None` after a restart.

There's a simple trick to migrate codecs in that case (see what's done in https://github.com/ACINQ/eclair/pull/1768 where I have the same kind of needs). However I'm not sure we want to go in that direction for this issue, give me some time to review this PR more in depth before we go further.

Reply:
I know but I think it's fine not to save it. Anytime we restart, this value will be ignored and overwritten anyway.

Reply:
You're completely right, I really didn't look hard enough at this PR, we only care about the `channelUpdate` field being stored!
Let's investigate the other changes proposed by pm47 though, we'll come back to this option later if we can't find a more satisfying one.